### PR TITLE
Add whole_word_jumping xontrib. Implements #2403

### PR DESCRIPTION
**Pull request by nh2**

### CODE REVIEW

File: xontrib/whole_word_jumping.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""Jumping across whole words (non-whitespace) with Ctrl+Left/Right.
+
+Alt+Left/Right remains unmodified to jump over smaller word segments.
+"""
+from prompt_toolkit.keys import Keys
+
```
Comment (astronouth7303) [inline xontrib/whole_word_jumping.py]: Please add an empty `__all__` to not polute the shell context.

**Comment by astronouth7303** [comment]
This is solid work. Thank you!

For the record, I debated requesting this be an external xontrib, but I think this is pretty useful, and is also a fine example of working with ptk in xontribs.

File: xontrib/whole_word_jumping.py (hunk 1 above)
Comment (nh2) [inline xontrib/whole_word_jumping.py]: Done.

**Comment by nh2** [comment]
> EDIT: Please do include a "new xontrib" news item.

Done.

**Comment by scopatz** [comment]
This looks good to me!

**Comment by astronouth7303** [comment]
What's up with the failing build? @gforsyth is this docs related?

**Comment by gforsyth** [comment]
Seems to be similar to the problem in #2408 -- something in the install is touching `xonsh/__init__.py` -- is there something added to amalgamation that we aren't aware of? 

And if there is, we _really_ need a test that isn't a failing docs build to catch that...

**Comment by scopatz** [comment]
I think we can go ahead and merge, right?

**Comment by astronouth7303** [comment]
I think so. The failed test is Travis's Linux/Py3.6, which I believe is also covered by CircleCI.

**Pull request merged**